### PR TITLE
Parse `BlockId` on forges side

### PR DESCRIPTION
**Pull request by piotmag769**
Tests work locally when running scarb built from commit [6d8703abbb9fc7e0e0a4b4c19ba73dd3b04198cb](https://github.com/software-mansion/scarb/commit/6d8703abbb9fc7e0e0a4b4c19ba73dd3b04198cb)
## Introduced changes

- we now get `Strings` instead of `BlockId` from scarb artifacts:
  - we use `ForkParamsString` instead of `RawForkParams` and `ForkConfigString` instead of `RawForkConfig`

## Checklist

- [x] Linked relevant issue
- [x] Updated relevant documentation
- [x] Added relevant tests
- [x] Performed self-review of the code
- [x] Added changes to `CHANGELOG.md`

### CODE REVIEW

File: crates/forge/src/scarb/config.rs [hunk 1]
```diff
@@ -97,29 +95,25 @@ impl TryFrom<RawForgeConfig> for ForgeConfig {
...
+                raw_fork_target.block_id.iter().collect_vec()[..]
+            else {
                 unreachable!()
             };
 
+            let block_id_type = match block_id_type.as_str() {
+                "number" => "Number",
+                "hash" => "Hash",
+                "tag" => "Tag",
+                _ => unreachable!(),
```
Comment (piotmag769) [inline crates/forge/src/scarb/config.rs:108]: May look weird, but I want the `unreachable!` arm https://en.wikipedia.org/wiki/Defensive_programming#Offensive_programming

File: crates/forge/src/scarb/config.rs [hunk 2]
```diff
@@ -79,7 +77,7 @@ pub(super) fn validate_raw_fork_config(raw_config: &RawForgeConfig) -> Result<()
         };
 
         if !["number", "hash", "tag"].contains(&&**block_id_key) {
-            bail!("block_id = {block_id_key} is not valid. Possible values = are \"number\", \"hash\" and \"tag\"");
+            bail!("block_id = {block_id_key} is not valid. Possible values are = \"number\", \"hash\" and \"tag\"");
```
Comment (piotmag769) [inline crates/forge/src/scarb/config.rs:80]: Ik it's unrelated

File: crates/forge/src/compiled_raw.rs [hunk 3]
```diff
@@ -30,24 +34,36 @@ pub enum CrateLocation {
 }
 
 #[derive(Debug, Clone, PartialEq, Deserialize)]
-pub enum RawForkConfig {
```
Comment (war-in) [inline crates/forge/src/compiled_raw.rs:33]: What was wrong with `raw` in the name?

File: crates/forge/src/scarb/config.rs (hunk 1 above)
Comment (war-in) [reply crates/forge/src/scarb/config.rs:108]: So, maybe we shouldn't handle it here?

File: crates/forge/src/lib.rs [hunk 4]
```diff
@@ -23,7 +23,7 @@ fn replace_id_with_params(
     fork_targets: &[ForkTarget],
 ) -> Result<RawForkParams> {
     match raw_fork_config {
-        RawForkConfig::Params(raw_fork_params) => Ok(raw_fork_params),
+        RawForkConfig::Params(fork_params_string) => Ok(fork_params_string),
```
Comment (war-in) [inline crates/forge/src/lib.rs:26]: unrelated?

File: crates/forge/src/scarb/config.rs (hunk 1 above)
Comment (piotmag769) [reply crates/forge/src/scarb/config.rs:108]: We decided to allow both lowercase and title case names

**Review by war-in — APPROVED** [review]

**Review by cptartur — APPROVED** [review]

File: crates/forge/src/compiled_raw.rs [hunk 5]
```diff
@@ -38,16 +42,28 @@ pub enum RawForkConfig {
...
 }
 
 impl TryFrom<RawForkParams> for ValidatedForkConfig {
     type Error = anyhow::Error;
 
     fn try_from(value: RawForkParams) -> Result<Self, Self::Error> {
+        let block_id = match value.block_id_type.to_lowercase().as_str() {
+            "number" => BlockId::Number(value.block_id_value.parse().unwrap()),
+            "hash" => BlockId::Hash(
+                Felt252::from(value.block_id_value.parse::<BigInt>().unwrap()).to_field_element(),
```
Comment (Arcticae) [inline crates/forge/src/compiled_raw.rs:56]: Wouldn't using our `conversions` be more appropriate here?

**Review by Arcticae — APPROVED** [review]

**Comment by piotmag769** [comment]
Will merge it together with #1157 with rebase, so that the PRs are in separate commits
